### PR TITLE
Fix schema generation warnings errors

### PR DESCRIPTION
**Pull request by rhysyngsun**
### What are the relevant tickets?
Fixes https://github.com/mitodl/hq/issues/4221

### Description (What does it do?)

This fixes outstanding errors and warnings and enforces that no new code will introduce new ones.

### Testing

- On `main`, verify you see warnings/errors when running `docker compose run --no-deps --rm web ./manage.py generate_openapi_spec`.
- On this branch, you should see no warnings with `docker compose run --no-deps --rm web ./manage.py generate_openapi_spec`

### CODE REVIEW

**Review by shanbady — CHANGES_REQUESTED** [review]
looks good. I tested a bunch of functionality where i thought some changes may interfere but it has not. Left a minor comment about something commented out in the Dockerfile. Also when i run the openapi generation script I saw two other warn lines 

```
 o.o.codegen.DefaultCodegen - The value (generator's option) must be either boolean or string. Default to `false`.
```
which i was able to resolve by adding this in scripts/generate_openapi.sh
```
--- a/scripts/generate_openapi.sh
+++ b/scripts/generate_openapi.sh
@@ -19,9 +19,9 @@ docker compose run --no-deps --rm web \
 GENERATOR_VERSION=v7.2.0
 
 docker run --rm -v "${PWD}:/local" -w /local openapitools/openapi-generator-cli:${GENERATOR_VERSION} \
-       generate -c scripts/openapi-configs/typescript-axios-v0.yaml
+       generate -c scripts/openapi-configs/typescript-axios-v0.yaml --additional-properties=supportsES6=false
 docker run --rm -v "${PWD}:/local" -w /local openapitools/openapi-generator-cli:${GENERATOR_VERSION} \
-       generate -c scripts/openapi-configs/typescript-axios-v1.yaml
+       generate -c scripts/openapi-configs/typescript-axios-v1.yaml --additional-properties=supportsES6=false
 
 # We expect pre-commit to exit with a non-zero status since it is reformatting
 # the generated code.
```


The other WARN was  
```
The 'exclude' field in hook 'check-added-large-files' is a regex, not a glob -- matching '/*' probably isn't what you want here
```

which I was able to suppress by modifying .pre-commit-config.yaml like so 
```
--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -16,7 +16,7 @@ repos:
         exclude: ".hbs$"
       - id: check-yaml
       - id: check-added-large-files
-        exclude: "yarn.lock|.yarn/releases/*|frontends/.yarn/releases/*"
+        exclude: "yarn.lock|.yarn/releases/.*|frontends/.yarn/releases/.*"
       - id: check-merge-conflict
       - id: check-toml
       - id: debug-statements
```

File: Dockerfile [hunk 1]
```diff
@@ -51,7 +51,7 @@ USER root
 COPY . /src
 WORKDIR /src
 RUN mkdir /src/staticfiles
-RUN chown -R mitodl:mitodl /src
+# RUN chown -R mitodl:mitodl /src
```
[the remark this answers is not in the document]
Comment (rhysyngsun) [reply Dockerfile:54]: This line was causing my `docker compose build`s to take nearly 30 minutes, so I removed it to speed things up. We end up mounting it as a volume anyway so I'm inclined to leave it out.

**Comment by rhysyngsun** [comment]
@shanbady addressed your feedback

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (shanbady) [reply Dockerfile:54]: Ok. we should probably just remove the commented line altogether then

**Comment by rhysyngsun** [comment]
@shanbady removed that line

**Review by shanbady — APPROVED** [review]

**Pull request merged**